Source/WebKit/WebKit.xcodeproj/project.pbxproj

```diff
@@ -17571,6 +17578,26 @@
 			shellPath = /bin/sh;
 			shellScript = "if [ \"${ACTION}\" = \"installhdrs\" -o \"${ACTION}\" = \"installapi\" ]; then\n    exit 0;\nfi\n\"${SCRIPT_INPUT_FILE_2}\"\n";
 		};
+		8473A286294D387E0051378B /* Unifdef Private Module Map */ = {
+			isa = PBXShellScriptBuildPhase;
+			buildActionMask = 2147483647;
+			files = (
+			);
+			inputFileListPaths = (
+			);
+			inputPaths = (
+				"$(PROJECT_DIR)/$(WK_MODULEMAP_PRIVATE_FILE)",
+			);
+			name = "Unifdef Private Module Map";
+			outputFileListPaths = (
+			);
+			outputPaths = (
+				"$(TARGET_BUILD_DIR)/$(MODULES_FOLDER_PATH)/module.private.modulemap",
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "HAS_WEBEVENTREGION=0\nfor includedSourceFileName in $INCLUDED_SOURCE_FILE_NAMES\ndo\n    if [ \"$includedSourceFileName\" = \"WebEventRegion.h\" ]\n    then\n        HAS_WEBEVENTREGION=1\n    fi\ndone\n\nif [ $HAS_WEBEVENTREGION -eq 0 ]\nthen\n    moduleMapUnifdefFlags=\"-UHAS_WEBEVENTREGION\"\nelse\n    moduleMapUnifdefFlags=\"-DHAS_WEBEVENTREGION\"\nfi\n\nunifdef $moduleMapUnifdefFlags -o \"$SCRIPT_OUTPUT_FILE_0\" \"$SCRIPT_INPUT_FILE_0\"\n\nreturnValue=$?\nif [ \\( $returnValue -ne 0 \\) -a \\( $returnValue -ne 1 \\) ]\nthen\n    exit $returnValue\nfi\n";
```

Comment:
I'm not sure if you want this pulled into Scripts instead. It probably has some style issues, though I tried to mostly match how some of the other scripts look.
```
HAS_WEBEVENTREGION=0
for includedSourceFileName in $INCLUDED_SOURCE_FILE_NAMES
do
    if [ "$includedSourceFileName" = "WebEventRegion.h" ]
    then
        HAS_WEBEVENTREGION=1
    fi
done

if [ $HAS_WEBEVENTREGION -eq 0 ]
then
    moduleMapUnifdefFlags="-UHAS_WEBEVENTREGION"
else
    moduleMapUnifdefFlags="-DHAS_WEBEVENTREGION"
fi

unifdef $moduleMapUnifdefFlags -o "$SCRIPT_OUTPUT_FILE_0" "$SCRIPT_INPUT_FILE_0"

returnValue=$?
if [ \( $returnValue -ne 0 \) -a \( $returnValue -ne 1 \) ]
then
    exit $returnValue
fi
```

Reply:
It probably is better as a separate script, but I'm not sure how important this is.

Reply:
I think it would be easier / more in line with project conventions to do this in WebKit's `DerivedSources.make`. From inside that Makefile, we have access to platform macros like `ENABLE(IOS_TOUCH_EVENTS)`, so we don't have to piggy-back off the INCLUDED_SOURCE_FILE_NAMES variable, and WebKit developers will better recognize the platform flag we're basing this off of.

Plus, it would mean we can use the normal MODULEMAP_PRIVATE_FILE build setting, instead of having to install the modulemap into the right place ourselves.

I played around a bit and came up with:
```makefile
PRIVATE_MODULEMAPS = \
	Modules/iOS_Private.modulemap \
	Modules/MacCatalyst_Private.modulemap \
	Modules/OSX_Private.modulemap \
#

GENERATED_MODULEMAPS = $(notdir $(PRIVATE_MODULEMAPS))

%.modulemap : Modules/%.modulemap
	unifdef -B $(addprefix -D, $(FEATURE_AND_PLATFORM_DEFINES)) -o $@ $<

all : $(GENERATED_MODULEMAPS)
```

This generates files like:
```
WebKitBuild/Debug/DerivedSources/WebKit/MacCatalyst_Private.modulemap
WebKitBuild/Debug/DerivedSources/WebKit/OSX_Private.modulemap
WebKitBuild/Debug/DerivedSources/WebKit/iOS_Private.modulemap
```

which can be accessed using `$(BUILT_PRODUCTS_DIR)/DerivedSources` in xcconfig.

If you do this, the conditional block in the modulemap should be:
```
#ifdef ENABLE_IOS_TOUCH_EVENTS
  explicit module WebEventRegion {
    header "WebEventRegion.h"
    export *
  }
#endif
```

Reply:
Spiffy, let me give that a try

Reply:
Xcode requires that the module map files already be present at the start of a build, so we still can't use `MODULEMAP_PRIVATE_FILE`. Is there a reasonable way for DerivedSources to install the module map files in the framework?

Reply:
Aw, too bad. Best I can think of is:

- Add the modulemap to WebKit's "Derived Sources" group. Easiest way to do this is to generate it once, and then use `Add Files to "WebKit"…` on the group.
- Create a Copy Files phase that copies it to "Wrapper", subpath `$(WK_FRAMEWORK_VERSION_PREFIX)Modules` (or MODULES_FOLDER_PATH, but not sure if that works in this context)

Reply:
We'd probably need to have Make name the output file `module.private.modulemap` in all cases, so like:
```make
ifneq ($(findstring iphone,$(WK_PLATFORM_NAME)),)
module.private.modulemap : Modules/iOS_Private.modulemap
else ifeq (maccatalyst,$(WK_PLATFORM_NAME))
module.private.modulemap : Modules/MacCatalyst_Private.modulemap
else ifeq (macosx,$(WK_PLATFORM_NAME))
module.private.modulemap : Modules/OSX_Private.modulemap
endif

%.modulemap : 
	unifdef -B $(addprefix -D, $(FEATURE_AND_PLATFORM_DEFINES)) -o $@ $<

all : module.private.modulemap
```

Reply:
👍 let me give that a spin

Reply:
You can kind of do "Products Directory" + `MODULES_FOLDER_PATH`, but it's technically wrong since you really want `$(TARGET_BUILD_DIR)/$(MODULES_FOLDER_PATH)` and "Products Directory" + `MODULES_FOLDER_PATH` is actually `$(BUILT_PRODUCTS_DIR)/$(MODULES_FOLDER_PATH)` So "Wrapper" + `$(WK_FRAMEWORK_VERSION_PREFIX)Modules` is regrettably what we need to do ("Wrapper" is `$(TARGET_BUILD_DIR)/$(WRAPPER_NAME)`)